xray/segment.go

```diff
@@ -68,6 +75,40 @@ func BeginSegment(ctx context.Context, name string) (context.Context, *Segment)
 		seg.GetService().Version = seg.ParentSegment.GetConfiguration().ServiceVersion
 	}
 
+	// Sampling strategy fallbacks to default sampling in the case of directly using BeginSegment API without any instrumentation
+	if r == nil && traceHeader == nil {
```

Comment:
I think this condition can be a logical or (`||`). If, for example, a customer calls this API and leaves one of the parameters nil we still can't infer a sampling decision and thus should fall back to the default sampling strategy.

Reply:
good catch!